refinery/tool_manager/utils.py

```diff
@@ -375,3 +427,56 @@ def get_visualization_annotations_list():
             visualization_annotations.append(json.loads(f.read()))
 
     return visualization_annotations
+
+
+def parse_file_relationship_nesting(nested_structure, nesting_dict=None,
+                                    nesting_level=0):
+    """
+    Recursive method to ensure that a ToolLaunchConfiguration's
+    `file_relationships` string is a proper representation of a LIST/PAIR
+    structure
+    :raises: RuntimeError if an inappropriately configured `file_relationships`
+     string is detected
+    """
+
+    if nesting_dict is None:
+        nesting_dict = {
+            nesting_level: {
```

Comment:
@hackdna I'm not changing the value of `nesting_level` here. It is still an integer.